mixer/pkg/log/wrapper.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"go.uber.org/zap/zapcore"
+)
+
+// wrapper is a simple wrapper around zapcore.Core which is used to inject
+// custom handling of level enabling/disabling.
+type wrapper struct {
+	original zapcore.Core
+	options  *Options
+}
+
+func (c *wrapper) Enabled(level zapcore.Level) bool {
+	switch level {
+	case zapcore.DebugLevel:
+		return c.options.OutputDebugging
+	case zapcore.ErrorLevel:
+		return c.options.OutputErrors
+	case zapcore.WarnLevel:
+		return c.options.OutputWarnings
+	case zapcore.InfoLevel:
+		return c.options.OutputInfo
+	}
+	return true
+}
+
+func (c *wrapper) With(fields []zapcore.Field) zapcore.Core {
```

Comment:
how do i gain access to this for use? I think we want the ability to create new loggers using `With()` in a few places, but it appears that you can't actually get a logger with the current code (outside of the log package). Is that correct?

Reply:
Correct, right now this is hidden.

My thinking was to keep the initial API surface simple and add stuff as we need it. 

We could just expose zap completely as-is and not provide any abstraction, aside maybe for standard Cobra flags. This would be simpler on our side, but I'm afraid the lack of opinion might lead to continued logging inconsistencies.